tests/unit/test_query_language.py

```diff
@@ -381,3 +384,36 @@ def test_incompatible_duration_operations(lhs, op, rhs):
         else:
             assert False
         assert not result
+
+
+@pytest.mark.parametrize(
+    "fn_name",
+    {
+        k
+        for k, v in DateFunctions.__dict__.items()
+        if callable(v) and "add_" not in k and "sub_" not in k
+    }
+    | {
+        k
+        for k, v in BaseSeries.__dict__.items()
+        if callable(v) and "__" not in k and "map" not in k
+    },
+)
+def test_ehrql_date_string_equivalence(fn_name):
+    date_str = "2000-01-01"
+    date_date = date(2000, 1, 1)
+
+    @table
+    class p(PatientFrame):
+        d = Series(date)
+
+    f = getattr(p.d, fn_name)
+    n_params = len(signature(f).parameters)
+    date_args = [date_date for i in range(n_params)]
+    str_args = [date_str for i in range(n_params)]
+
+    if "in" in fn_name:
```

Comment:
This is might be quite brittle.  What happens if we add a new method called eg `contains`?

I think it'd be better to be explicit: `if fn_name in ["is_in", "is_not_in"]`.  In addition to being more robust, this makes the reader do less thinking.